include/deal.II/base/point.h

```diff
@@ -501,6 +514,33 @@ Point<dim, Number>::operator=(const Tensor<1, dim, OtherNumber> &p)
 
 
 
+template <int dim, typename Number>
+inline DEAL_II_CONSTEXPR DEAL_II_CUDA_HOST_DEV Point<dim + 1, Number>
+                                               Point<dim, Number>::extend(const Number extension) const
+{
+  Point<dim + 1, Number> result;
+  for (unsigned int i = 0; i < dim; ++i)
+    result[i] = this->values[i];
+  result[dim] = extension;
+  return result;
+}
+
+
+
+template <int dim, typename Number>
+inline DEAL_II_CONSTEXPR DEAL_II_CUDA_HOST_DEV Point<dim - 1, Number>
+                                               Point<dim, Number>::project(const unsigned int coord_axis) const
+{
+  Point<dim - 1, Number> result;
+  for (unsigned int i = 0; i < coord_axis - 1; ++i)
```

Comment:
This won't work if `coord_axis==0` since `unsigned int (-1)` is much larger than any reasonable `dim` here. Same below.

Reply:
Thank you for noticing it. I fixed it.